R/ci.R

```diff
@@ -1,82 +1,104 @@
 #' Continuous integration setup and badges
 #'
 #' Sets up continuous integration (CI) services for an R package that is
-#' developed on GitHub. CI services can run `R CMD check` automatically on
-#' various platforms, triggered by each push or pull request. These functions
-#' * Add service-specific configuration files and add them to `.Rbuildignore`.
-#' * Activate a service or give the user a detailed prompt.
-#' * Provide the markdown to insert a badge into README.
+#' developed on GitHub with CI-agnostic support by \pkg{tic}.
+#' CI services can run `R CMD check` automatically on
+#' various platforms, triggered by each push or pull request. This function
+#' * Adds service-specific configuration files and adds them to `.Rbuildignore`.
+#' * Activates a service or gives the user a detailed prompt.\cr
 #'
-#' @name ci
-#' @aliases NULL
-NULL
-
-#' @section `use_travis()`:
-#' Adds a basic `.travis.yml` to the top-level directory of a package. This is a
-#' configuration file for the [Travis CI](https://travis-ci.org/) continuous
-#' integration service.
-#' @param browse Open a browser window to enable automatic builds for the
-#'   package.
+#' @details
+#' By default the CI-services "Travis" (Linux) and "Appveyor"
+#' (Windows) will be set up. Basic `.travis.yml` and `appveyor.yml` files are
+#' added to the top-level directory of a package.
+#'
+#' This function is aimed at supporting the most common use cases.
+#' Users who require more control are advised to manually call the individual
+#' functions.
+#'
+#' @name use_ci
+#' @importFrom travis travis_set_pat
+#' @param path `[string]`\cr
+#'   The path to the repo to prepare.
+#' @param quiet `[flag]`\cr
+#'   Less verbose output? Default: `FALSE`.
+#' @param services `[character]`\cr
+#'   CI services to add.
 #' @export
 #' @rdname ci
-use_travis <- function(browse = interactive()) {
-  check_uses_github()
-
-  use_template(
-    "travis.yml",
-    ".travis.yml",
-    ignore = TRUE
-  )
+use_ci <- function(path = ".", quiet = FALSE,
+                   services = c("Travis", "Appveyor")) {
+  #' @details
+  #' The following steps will be run:
+  withr::with_dir(path, {
+    #' 1. If necessary, create a GitHub repository via [use_github()]
+    travis:::use_github_interactive() #FIXME: Export function in travis
+    stopifnot(uses_github())
+
+    if ("Travis" %in% services) {
+      #' 1. Enable Travis via [travis_enable()]
+      travis_enable()
+
+      #' 1. Create a default `.travis.yml` file
+      #'    (overwrite after confirmation in interactive mode only)
+      travis:::use_travis_yml() #FIXME: Export function in travis
+    }
 
-  travis_activate(browse)
-  use_travis_badge()
+    repo_type <- travis:::detect_repo_type() #FIXME: Export function in travis
```

Comment:
What functions do we need to make available in _travis_ for this PR? For some lower-level functions it's better to just copy/move them here.